src/reason_pprint_ast.ml

```diff
@@ -2388,10 +2388,7 @@ let is_unit_pattern x = match x.ppat_desc with
   | _ -> false
 
 let is_direct_pattern x = x.ppat_attributes = [] && match x.ppat_desc with
-  | Ppat_array _ | Ppat_record _
-  | Ppat_construct ( {txt= Lident"()"}, None)
-  | Ppat_construct ( {txt= Lident"[]"}, None)
-  | Ppat_construct ( {txt= Lident"::"}, Some _) -> true
+  | Ppat_construct ( {txt= Lident"()"}, None) -> true
```

Comment:
@IwanKaramazow - you mentioned removing the two cases added in a214071eba5e6ba72946c1655dc63a650f2bf7f1, but I found that Ppat_array and Ppat_record also cause the same inconsistencies. This will print all of the above with parens, but still allow typing without.

Reply:
totally forgot about arrays and records 😅